### DISCUSSION TITLE
Database connection issue

### DISCUSSION THREAD

**Discussion by BelIblis31** [tillsteinbach/VWsFriend#191] [Q&A]
Hello,

I face the following error trying to install the packages with docker:

vincent-postgresdb-1  | 2022-03-11 15:35:39.894 UTC [40] FATAL:  password authentication failed for user "admin"
vincent-postgresdb-1  | 2022-03-11 15:35:39.894 UTC [40] DETAIL:  Password does not match for user "admin".
vincent-postgresdb-1  |         Connection matched pg_hba.conf line 99: "host all all all md5"
vincent-vwsfriend-1   | 2022-03-11T15:35:39+0000:ERROR:agent_connector:Could not establish a connection to database, will try again after 10 seconds: (psycopg2.OperationalError) connection to server at "postgresdbbackend" (172.19.0.2), port 5432 failed: FATAL:  password authentication failed for user "admin"

Password is correctly set in the ENV file but it seems the user creation didn’t went well.
Do you have any suggestion ?

**Comment by tillsteinbach** [comment]
what does your startup command look like?

**Reply by BelIblis31** [reply]
sudo docker-compose --env-file ./myconfig.env up
I use the yml file from github source and here my env file (I removed passwords):

GF_SECURITY_ADMIN_USER=admin_gf
GF_SECURITY_ADMIN_PASSWORD=
GF_SERVER_HTTP_PORT=4001
DB_USER=admin
DB_PASSWORD=
DB_PORT=5432
VWSFRIEND_USERNAME=admin
VWSFRIEND_PASSWORD=
WECONNECT_USER=
WECONNECT_PASSWORD=
WECONNECT_INTERVAL=300
HOMEKIT_INTERFACE=eth0
HOMEKIT_IP=192.168.0.234
HOMEKIT_MASK=192.168.0.0/24
HOMEKIT_GW=192.168.1.1
DATABASE_URL=postgresql://${DB_USER-admin}:${DB_PASSWORD-secret}@${DB_HOSTNAME-postgresdbbackend}:${DB_PORT-5432}/${DB_NAME-vwsfriend}
POSTGRES_DB=vwsfriend
TZ=UTC
PGTZ=UTC

**Reply by tillsteinbach** [reply]
Do you have "very" special symbols in the DB_PASSWORD?

**Reply by tillsteinbach** [reply]
Ahhh, and why do you have the DATABASE_URL in your ENV file? I think what might maybe go wrong is the variable substitution in DATABASE_URL

**Reply by BelIblis31** [reply]
> Do you have "very" special symbols in the DB_PASSWORD?

I had first but removed them (especially @ one !)

> Ahhh, and why do you have the DATABASE_URL in your ENV file? I think what might maybe go wrong is the variable substitution in DATABASE_URL

Not sure where I got that. I just tried removing that line but I get the same error message

**Reply by BelIblis31** [reply]
Full errors lines:

vincent-postgresdb-1  | 2022-03-11 16:49:49.770 UTC [68] FATAL:  password authentication failed for user "admin"
vincent-postgresdb-1  | 2022-03-11 16:49:49.770 UTC [68] DETAIL:  Password does not match for user "admin".
vincent-postgresdb-1  |         Connection matched pg_hba.conf line 99: "host all all all md5"
vincent-vwsfriend-1   | 2022-03-11T16:49:49+0000:ERROR:agent_connector:Could not establish a connection to database, will try again after 10 seconds: (psycopg2.OperationalError) connection to server at "postgresdbbackend" (172.19.0.2), port 5432 failed: FATAL:  password authentication failed for user "admin"
vincent-vwsfriend-1   |
vincent-vwsfriend-1   | (Background on this error at: https://sqlalche.me/e/14/e3q8)

**Reply by tillsteinbach** [reply]
I would start with only the variables in the env file that are in the template: https://raw.githubusercontent.com/tillsteinbach/VWsFriend/main/.env

**Reply by BelIblis31** [reply]
ok I just tried again and got the following (same) error:

vincent-postgresdb-1  | 2022-03-11 18:44:47.078 UTC [241] FATAL:  password authentication failed for user "admin"
vincent-postgresdb-1  | 2022-03-11 18:44:47.078 UTC [241] DETAIL:  Password does not match for user "admin".
vincent-postgresdb-1  |         Connection matched pg_hba.conf line 99: "host all all all md5"
vincent-vwsfriend-1   | 2022-03-11T18:44:47+0000:ERROR:agent_connector:Could not establish a connection to database, will try again after 10 seconds: (psycopg2.OperationalError) connection to server at "postgresdbbackend" (172.19.0.2), port 5432 failed: FATAL:  password authentication failed for user "admin"
vincent-vwsfriend-1   |
vincent-vwsfriend-1   | (Background on this error at: https://sqlalche.me/e/14/e3q8)

The only other ENV parameter I kept is GF_SERVER_HTTP_PORT=4001 because 3000 is already in use on my server.
Also I use this yml file : https://github.com/tillsteinbach/VWsFriend/blob/main/docker-compose.yml

**Reply by tillsteinbach** [reply]
Can you do a `docker-compose down -v` that will delete all volumes and all data. And then do the up command again? Maybe postgres needs to start again from zero to work with the changed password.

**Reply by BelIblis31** [reply]
It seems to have worked well !
However I can’t access anything either on 4000 or 3000 (4001 in my case) port. But I think it is completely different issue. 
Thanks a lot for your help and reactivity!
